hammer/sim/xcelium/__init__.py

```diff
@@ -427,35 +766,415 @@ def elaborate_xrun(self) -> bool:
       
     if xrun_opts["mce"]: elab_opts.append(self.generate_mc_cmd())
     elab_opts = ('ELABORATION', elab_opts)
+
+    
         
     arg_file_path = self.generate_arg_file("xrun_elab.arg", "HAMMER-GEN XRUN ELAB ARG FILE", [elab_opts])
     args =[self.xcelium_bin]
     args.append(f"-elaborate -f {arg_file_path}")
-    
+
     self.update_submit_options()
+    ### If AMS enabled, submit options but do not run elaborate sub-step.
+    if self.get_setting(f"{self.tool_config_prefix()}.ams"):
+      return True
+    """if self.get_setting(f"{self.tool_config_prefix()}.ams"):
+      ams_elab_opts = self.extract_ams_opts("elaborate")
+      self.attach_opts(arg_file_path, ams_elab_opts)
+      #Add all necessary AMS specific commands, remove other commands as necessary"""
+    
     self.run_executable(args, cwd=self.run_dir)
     return True
 
   def sim_xrun(self) -> bool:
     sim_opts  = self.extract_sim_opts()[1]
+    ams_opts = self.extract_ams_opts("sim")
     sim_cmd_opts = self.get_setting(f"{self.sim_input_prefix}.options", [])
     sim_opts_removal  = ["tb_name", "input_files", "incdir"]
     xrun_opts_removal = ["enhanced_recompile", "mce"]
+    ams_opts_removal = []
+
+    #if (self.get_setting(f"sim.xcelium.ams")): sim_cmd_opts.extend(ams_opts)
+
     sim_cmd_opts = ('SIMULATION', sim_cmd_opts)
     
     if not sim_opts["execute_sim"]:
       self.logger.warning("Not running any simulations because sim.inputs.execute_sim is unset.")
       return True
     
-    arg_file_path = self.generate_arg_file("xrun_sim.arg", "HAMMER-GEN XRUN SIM ARG FILE", [sim_cmd_opts],
+    
+
+    arg_file_path = self.generate_arg_file("xrun_sim.arg", "HAMMER-GEN XRUN SIM ARG FILE", [sim_cmd_opts, 'SIMULATION'],
                                            sim_opt_removal = sim_opts_removal,
                                            xrun_opt_removal = xrun_opts_removal)    
     args =[self.xcelium_bin]
     args.append(f"-R -f {arg_file_path} -input {self.sim_tcl_file}")
 
     self.generate_sim_tcl() 
     self.update_submit_options()
+
+    ### If AMS enabled, submit options and procede to AMS single-step function
+    if self.get_setting(f"{self.tool_config_prefix()}.ams"):
+      self.run_mxh_pseudo_three_step()
+      return True
+
+
     self.run_executable(args, cwd=self.run_dir)
     return True
 
+  
+  def sift_exts(list, exts):
```

Comment:
Ditto this for making general utility.
But more generally, in `hammer/utils/__init__.py`, there's already a `HammerFileType` Enum. I'd recommend doing this retreival/sifting and preparation methods below by just doing map/filter/reduce operations, where the partial function checks `HammerFileType.get_filetype(<file>) == HammerFileType.VERILOG` or similar. Feel free to expand the enum as necessary.